### PR TITLE
[TTAHUB-2100] Enable SES in non-production

### PR DESCRIPTION
**Pull request by kryswisnaskas**
## Description of change
This PR allows to enable AWS SES in the non-production environments by eliminating bounces to fake email addresses. It flags the generated emails with a "no-send_" prefix and filters out these addresses. It also prevents sending to duplicate emails.

## How to test
Verify on locally (while having the latest transformed db snapshot) or on Dev that notifications work as expected, e.g. by adding yourself as a collaborator.

Extra check: Verify there is no record of sending an email to a fake address by looking at the MailerLogs table.
e.g.

`SELECT * FROM "MailerLogs" order by "createdAt" desc LIMIT 10;`

## Issue(s)

* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-2100

### Every PR

- [x] Meets issue criteria
- [x] JIRA ticket status updated
- [x] Code is meaningfully tested
- [n/a] Meets accessibility standards (WCAG 2.1 Levels A, AA)
- [n/a] API Documentation updated
- [n/a] Boundary diagram updated
- [n/a] Logical Data Model updated
- [n/a] [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions
- [n/a] UI review complete

### Production Deploy

- [ ] Staging smoke test completed

### After merge/deploy

- [ ] Update JIRA ticket status

### CODE REVIEW

**Comment by kryswisnaskas** [comment]
Dev is currently not functioning due to the db. I'll this PR open, but testing would need to be done locally.

File: src/lib/mailer/index.js [hunk 1]
```diff
@@ -73,6 +73,21 @@ export const frequencyToInterval = (freq) => {
...
+ * Filters and deduplicates an array of email addresses, removing duplicates
+ * and excluding email addresses that start with 'no-send_'.
+ *
+ * @param {string[]} emails - An array of email addresses to filter and deduplicate.
+ * @returns {string[]} - A deduplicated and filtered array of email addresses.
+ */
+export const filterAndDeduplicateEmails = (emails) => {
+  const filteredEmails = emails.flat()
+    .filter((email) => typeof email === 'string' && !email.startsWith('no-send_'))
+    .filter((email, index, array) => array.indexOf(email) === index);
```
Comment (GarrettEHill) [inline src/lib/mailer/index.js:86]: Is this faster then building an serializing a set?

Comment (kryswisnaskas) [reply src/lib/mailer/index.js:86]: I initially began with a set implementation but then switched to filtering. This method is one of the simplest ways to achieve both filtering and deduplication. While its performance implications are uncertain, considering our dataset consists of a very small number of email addresses, any performance difference should be negligible.

**Comment by kryswisnaskas** [comment]
> Dev is currently not functioning due to the db. I'll this PR open, but testing would need to be done locally.

This is no longer the case. Dev is working and had this branch pushed to it.

File: frontend/src/fetchers/users.js [hunk 2]
```diff
@@ -17,8 +17,8 @@ export const getActiveUsers = async () => {
   return res.blob();
 };
 
-export const getTrainingReportUsers = async (regionId, eventId) => {
-  const res = await get(join('/', 'api', 'users', 'training-report-users', `?regionId=${String(regionId)}&eventId=${String(eventId)}`));
+export const getTrainingReportUsers = async (regionId) => {
```
Comment (thewatermethod) [inline frontend/src/fetchers/users.js:20]: I'm seeing changing that look they are from another PR here - bad merge?

**Comment by thewatermethod** [comment]
👍 Tested locally and on dev, notifications work

Should we clean up the merge before putting this PR in main?

**Comment by kryswisnaskas** [comment]
> 👍 Tested locally and on dev, notifications work
> 
> Should we clean up the merge before putting this PR in main?

Closing this PR due to issues with a complex merge and opening a new one https://github.com/HHS/Head-Start-TTADP/pull/1748

**Pull request closed without merging**